library/core/src/convert/mod.rs

```diff
@@ -22,6 +22,15 @@
 //! may be necessary to implement [`Into`] or [`TryInto`] directly when converting to a type
 //! outside the current crate.
 //!
+//! # When to implement these traits
+//!
+//! Generally, the conversion traits in this module are intended to convert between types that
+//! are semantically equivalent and differ only in representation.
+//!
+//! For example, [`From`]/[`TryFrom`] implementations exist for string to string, list to list,
+//! number to number, or error to error conversions. However, implementing `From<i32>` for
+//! `String` or `TryFrom<&str>` for `i32` would not be appropriate.
```

Comment:
```suggestion
//! number to number, or error to error conversions, as well as conversions between
//! `OwnedFd` or `OwnedHandle` and types such as `File`.
//!
//! However, don't use `From` or `TryFrom` for:
//! - Formatting, such as implementing `From<i32>` for `String`; use [`Display`] for that.
//! - Parsing, such as implementing `TryFrom<&str>` for `i32`; use [`FromStr`] for that.
```

(Needs word-wrapping.)